Dockerfile

```diff
@@ -1,23 +1,36 @@
-FROM openjdk:8-jdk
+FROM maven:3.6.3-jdk-8 as build
 
 ENV JAVA_OPTS "-server -Xconcurrentio -Xmx1g -Xms1g -XX:+UseG1GC -Ddw.server.application_connectors[0].bind_host=0.0.0.0 -Ddw.server.application_connectors[0].port=8989"
 
-RUN mkdir -p /data && mkdir -p /graphhopper
-
 # install node - only required for JS UI
 RUN apt-get install -y wget \
        && curl -sL https://deb.nodesource.com/setup_13.x | bash - \
        && apt-get install -y nodejs
 
-COPY . /graphhopper/
-
 WORKDIR /graphhopper
 
+COPY . .
+
 # create main.js - only required for JS UI
-RUN cd web && npm install && npm run bundleProduction && cd ..
+WORKDIR /graphhopper/web
+
+RUN npm install && npm run bundleProduction
+
+WORKDIR /graphhopper
 
 RUN ./graphhopper.sh build
 
+FROM openjdk:11.0.5-jre
+
+ENV JAVA_OPTS "-server -Xconcurrentio -Xmx1g -Xms1g -XX:+UseG1GC -Ddw.server.application_connectors[0].bind_host=0.0.0.0 -Ddw.server.application_connectors[0].port=8989"
+
+RUN mkdir -p /data
+
+WORKDIR /graphhopper
+
+COPY --from=build /graphhopper/web/target/*.jar ./web/target/
+COPY ./graphhopper.sh ./pom.xml ./config-example.yml ./
```

Comment:
Is the pom.xml file really necessary here? As we use a jre (second FROM) compiling is not possible anymore.

Reply:
It is used to get the version in the bash script. Changing the bash script will allow removing this file from the docker file if the version will be fetched from other location or getting the jar file in another method.
My initial attempt where without it, but this is the simplest solution for the docker.
I think the version in the bash file to find the right jar file should use the jar file name instead, but it is fragile and I think out of this scope..

Reply:
Ah, thanks for finding this out. Can you add a comment and point to this issue on why these files need to be in the Docker image?

Reply:
Done.

Reply:
Somehow there are 0 commits in this PR now. Can you have a look?

Reply:
I'm not sure what happened, some automatic merge of some sort. Should be OK now...